libqtile/layout/base.py

```diff
@@ -293,19 +293,34 @@ def focus_previous(self, win):
             return self[idx - 1]
         return None
 
-    def add(self, client, offset_to_current=0):
-        """
-        Insert the given client into collection at position of the current.
-
-        Use parameter 'offset_to_current' to specify where the client shall be
-        inserted. Defaults to zero, which means at position of current client.
-        Positive values are after the client.
-        """
-        pos = max(0, self._current_idx + offset_to_current)
-        if pos < len(self.clients):
-            self.clients.insert(pos, client)
+    def add(self, client, offset_to_current=0, client_position=None):
+        """
+        If client_position is set by user, insert the given client depending of
+          its value : top, bottom, after_current, before_current.
+        Else:
+          Insert the given client into collection at position of the current.
+
+          Use parameter 'offset_to_current' to specify where the client shall be
+          inserted. Defaults to zero, which means at position of current client.
+          Positive values are after the client.
```

Comment:
It's not a big deal but I don't like the if-else structure and indentation. Let's write plain English, not pseudo-code. :)

(also, let's avoid modifying a line if we can)